### DISCUSSION TITLE
`ModuleNotFoundError` when running Python

### DISCUSSION THREAD

**Discussion by dynamite-bud** [wasmerio/wasmer#4248] [Q&A]
### Describe the bug

The problem is that the default current directory is `/`. Python gets its standard library from the files embedded in the `python/python` package, which are initially mounted in their default spot. When mounting the host's current directory to the guest's current directory, we end up clobbering the existing filesystem and core modules like `encodings` will no longer be available.

### Steps to reproduce

```shell
$ wasmer run python/python --mapdir=.:. test.py
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
Fatal Python error: Py_Initialize: Unable to get the locale encoding
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00000000 (most recent call first):
error: RuntimeError: unreachable
    at <unnamed> (<module>[4661]:0x224edf)
    at <unnamed> (<module>[2033]:0xce8da)
    at <unnamed> (<module>[2030]:0xcde7a)
    at <unnamed> (<module>[2028]:0xcd9d0)
    at <unnamed> (<module>[2037]:0xcef38)
    at <unnamed> (<module>[43]:0x4743)
    at <unnamed> (<module>[42]:0x3fec)
    at <unnamed> (<module>[4709]:0x22985e)
    at <unnamed> (<module>[30]:0x3e58)
```

### Additional context

The issue isn't specific to the python wapm package

**Comment by dynamite-bud** [comment]
The fix (from @Michael-F-Bryan ) is to mount the current directory on the host to a different directory in the guest (i.e. run `wasmer run python/python --mapdir=/app:. /app/test.py`).

**Comment by dynamite-bud** [comment]
We can mitigate the user experience impact. some ideas from @waynr :
1. Wasmer could warn users when they have overlapping directory mappings
2. Wasmer could refuse to overlap directory mappings (especially the environment's root directory) without some kind of --force-overlapped-mappings flag
3. Reconsider for a future breaking change using a different default working directory

**Comment by miraclejzd** [comment]
Hi，are there any solutions? I am struggling with some python dependencies questions.

**Reply by theduke** [reply]
@miraclejzd what exactly are you struggling with?

Can you give an example?

In general, you'd need to mount your dependencies somewhere (--mapdir) and then amend the PYTHONPATH env var with the location.
